sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -344,6 +344,8 @@ case class MapFilter(
     Examples:
       > SELECT _FUNC_(array(1, 2, 3), x -> x % 2 == 1);
        [1,3]
+      > SELECT _FUNC_(array(0, 2, 3), (x, i) -> x > i);
+       [2, 3]
   """,
   since = "2.4.0")
```

Comment:
Could you add a `note` to describe this can take the index argument since 3.0.0? E.g., from `collectionOperations.scala`:

https://github.com/apache/spark/blob/740293528edbae412a8accf1d2e8ddd79a49766d/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala#L1049-L1051